### PR TITLE
Update the TCK to 2.0.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HV-1046
 * https://hibernate.atlassian.net/browse/HV-1090
 * Update remaining references to BV 1.1

### CODE REVIEW

**Comment by gunnarmorling** [comment]
No biggie, but can we try and reduce the number of issue-less commits? E.g. https://github.com/hibernate/hibernate-validator/pull/488/commits/07f8f830368ad3b5b33e8b58f22defdede1133ed could be assigned to [HV-1085](https://hibernate.atlassian.net/browse/HV-1085).

File: documentation/src/main/asciidoc/ch11.asciidoc [hunk 1]
```diff
@@ -424,7 +424,7 @@ unwrapped to `HibernateConstraintViolation` in order to retrieve the dynamic pay
 ====
 [source, JAVA, indent=0]
 ----
-include::{sourcedir}/org/hibernate/validator/referenceguide/chapter11/dynamicpayload/DynamicPayloadTest.java[tags=include]
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter11/dynamicpayload/DynamicPayLoadTest.java[tags=include]
```
Comment (gunnarmorling) [inline documentation/src/main/asciidoc/ch11.asciidoc]: Ah, I fugure that one was broken? I just closed [HV-1046](https://hibernate.atlassian.net/browse/HV-1046) beause I couldn't it, but it seems that's the one? If so, let's re-open and reference it then.

**Comment by gunnarmorling** [comment]
Looks good overall, added some comments.

**Comment by gsmet** [comment]
Rebased and fixed.

**Pull request merged**